app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -898,29 +905,78 @@ class BrowserTabViewModel(
         ctaViewModel.registerDaxBubbleCtaDismissed(cta)
     }
 
-    fun onUserClickCtaOkButton() {
-        val cta = ctaViewState.value?.cta ?: return
+    fun onUserClickCtaOkButton(cta: Cta) {
```

Comment:
I found out some race conditions when using the current state from ViewModel instead of the CTA value from the original listener.

For example:
* We prompt a dialog
* User clicks in the ok button
* Dialog triggers `onUserClickCtaOkButton`
* New logic triggers a new dialog and CTA ViewState gets updated
* Dialog gets dismissed and triggers `onUserDismissedCta`
* The wrong dialog will be tracked (ViewState contains the new dialog rather than the previous one)

There are more scenarios, but this already shows that using this mutable ViewModel-ViewState could have non-expected behaviors. I think is more correct to rely on the actual value received from the listener.